src/components/ImpactDialog.js

```diff
@@ -0,0 +1,194 @@
+  const classes = useStyles()
+  let dialogContent
+  switch (modalType) {
+    case 'confirmImpact':
+      dialogContent = (
+        <>
+          <MuiDialogTitle disableTypography className={classes.root}>
+            <Typography variant="h6">
+              Are you ready to turn your Tabs into a force for good?
+            </Typography>
+          </MuiDialogTitle>
+          <MuiDialogContent>
+            <div className={classes.centerImage}>
+              <RandomGif />
+            </div>
+            <Typography className={classes.typographySpacing}>
+              Each time you open a tab, you'll be helping shelter cats get
+              adopted by{' '}
+              <a
+                target="_blank"
+                rel="noopener noreferrer"
+                href="https://thejacksongalaxyproject.greatergood.org/about/cat-pawsitive/"
+              >
+                providing treats to be used in positive reinforcement training.
+              </a>{' '}
+              Ready to get started?
+            </Typography>
+          </MuiDialogContent>
+          <MuiDialogActions>
+            <Button
+              autoFocus
+              onClick={buttonOnClick}
+              variant="contained"
+              color="primary"
+            >
+              I'M READY!
+            </Button>
+          </MuiDialogActions>
+        </>
+      )
+      break
+    case 'impactWalkMe':
+      dialogContent = (
+        <>
+          <MuiDialogTitle disableTypography className={classes.root}>
+            <Typography variant="h6">Open a new tab</Typography>
+          </MuiDialogTitle>
+          <MuiDialogContent className={classes.walkMe}>
+            <Typography id="alert-dialog-description">
+              When you do, you'll donate enough to give a shelter cat a treat
+              during training. We'll track how many treats you've given on the
+              top of the page:
+            </Typography>
+            <div className={classes.impactCounter}>
+              <ImpactCounter includeNumber number={0} progress={100} />
+            </div>
+          </MuiDialogContent>
+        </>
+      )
+      break
+    case 'claimImpactReward':
+      dialogContent = (
+        <>
+          <MuiDialogTitle disableTypography className={classes.root}>
+            <Typography variant="h6">
+              You just helped a shelter cat! Congrats!
+            </Typography>
+          </MuiDialogTitle>
+          <MuiDialogContent>
+            <div className={classes.centerImage}>
+              <RandomGif />
+            </div>
+            <Typography className={classes.typographySpacing}>
+              The more people who open tabs, the more good we can do! Get a
+              friend on board and we'll give a treat to another 10 shelter cats:
+            </Typography>
+            <div className={classes.shareContainer}>
+              <InviteFriends user={user} />
+              <SocialShare url="https://tab.gladly.io" />
```

Comment:
This should be the user's referral URL. Likely extract the URL constructor into a util and use it here + in InviteFriends.